torch/csrc/distributed/c10d/CUDASymmetricMemory-inl.h

```diff
@@ -289,8 +301,21 @@ __device__ __inline__ void multimem_st(T* mc_ptr, Vec<Alignment>& vec) {
 
 template <int Alignment, typename T>
 __device__ __inline__ Vec<Alignment> ld_vec(const T* addr) {
-#if defined(USE_ROCM) || (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 800))
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 800))
   CUDA_KERNEL_ASSERT(false);
+#elif defined(USE_ROCM)
+  Vec<Alignment> vec;
+  if constexpr (Alignment == 16) {
+    vec.u64[0] = reinterpret_cast<const uint64_t*>(addr)[0];
```

Comment:
does rocm have 128-bit stores?

Reply:
Yes, thanks for catching it! Will push a fix.